src/test/java/seedu/application/logic/commands/EditCommandTest.java

```diff
@@ -82,25 +82,6 @@ public void execute_noFieldSpecifiedUnfilteredList_success() {
         assertCommandSuccess(editCommand, model, expectedMessage, expectedModel);
     }
 
-    @Test
-    public void execute_filteredList_success() {
-        showApplicationAtIndex(model, INDEX_FIRST_APPLICATION);
-
-        Application applicationInFilteredList = model.getFilteredApplicationList()
-                .get(INDEX_FIRST_APPLICATION.getZeroBased());
-        Application editedApplication = new ApplicationBuilder(applicationInFilteredList)
-                .withCompany(VALID_COMPANY_FACEBOOK).build();
-        EditCommand editCommand = new EditCommand(INDEX_FIRST_APPLICATION,
-                new EditApplicationDescriptorBuilder().withCompany(VALID_COMPANY_FACEBOOK).build());
-
-        String expectedMessage = String.format(EditCommand.MESSAGE_EDIT_APPLICATION_SUCCESS, editedApplication);
-
-        Model expectedModel = new ModelManager(new ApplicationBook(model.getApplicationBook()), new UserPrefs());
-        expectedModel.setApplication(model.getFilteredApplicationList().get(0), editedApplication);
-
-        assertCommandSuccess(editCommand, model, expectedMessage, expectedModel);
-    }
-
```

Comment:
This test code is deleted for following reasons:

* The test code originally is to test the filtered application list will update to SHOW_ALL_PREDICATE after each EditCommand is execute.
* However, in archive feature, the list of applications is shown in two different way: archive list & normal list without archive.
* Hence, I think it will be better if user can edit application directly in archive list also. After the edit command is execute, the list shown in UI should be still remain in archive list.

Reply:
I get what you mean on the two different views - archive list & normal list - but I was wondering if it makes sense to allow edit command to work on the archive list? Since if something is archived, I would think that it shouldn't be edited in case there is some accidental overwrite. Do you think it would make more sense if the user wants to edit the an application in the archive list, then that application needs to be retrieved first?

Reply:
I think it will just be more convenient to user. But it is also possible to only allow user to edit on non-archived application.

As for the overwriting concern, it should be not a problem because the "archive list" is actually not a different list itself. It is just applying the `ShowArchiveOnlyPredicate` and `HideArchiveFromListPredicate` to the filter list and user still cannot add an application if the application is archived. If user edited the archived application same to one of the non-archived application, duplicate exception will also be thrown just like the normal usage of EditCommand.

But yeah I can change the edit to only non-archived application. Lets see what @jhchee18 or other members think @nmtuan2001 @teekaytai 🥲

Reply:
hmm yup if we take reference from the whatsapp example then I think this behaviour would work. I think the consideration on user convenience makes sense as well. In that case I think we should just stick to the existing behaviour 👍